service/src/main/java/tds/assessment/services/impl/AssessmentAssembler.java

```diff
@@ -3,57 +3,159 @@
 import com.google.common.collect.ArrayListMultimap;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import tds.assessment.Algorithm;
 import tds.assessment.Assessment;
 import tds.assessment.Form;
 import tds.assessment.Item;
 import tds.assessment.ItemConstraint;
 import tds.assessment.ItemProperty;
 import tds.assessment.Segment;
 import tds.assessment.Strand;
+import tds.assessment.repositories.FormQueryRepository;
+import tds.assessment.repositories.ItemQueryRepository;
 
 /**
  * A helper class for assembling the {@link tds.assessment.Assessment}
  */
 class AssessmentAssembler {
+    private final ItemQueryRepository itemQueryRepository;
+    private final FormQueryRepository formQueryRepository;
+    private ArrayListMultimap<String, ItemProperty> itemIdToPropertyMap;
+    private List<Item> adaptiveAssessmentItems;
+    private Map<String, List<Item>> fixedFormAssessmentItems;
+    private List<Form> fixedFormAssessmentForms;
+
+    AssessmentAssembler(ItemQueryRepository itemRepository, FormQueryRepository formRepository) {
```

Comment:
I agree with Todd's comments on the assembler containing repository calls. I don't think AssessmentAssembler should contain anything other that static methods as it should not hold state.